### PR TITLE
Feat/weapons

### PR DESCRIPTION
**Pull request by NathanGr33n**
Weapon system UI Displaying, no weapons displaying yet

### CODE REVIEW

File: TKGame/Content/Weapons/Sword.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    lock (syncRoot)
+                    {
+                        if (instance == null)
+                        {
+                            instance = new Sword();
```
Comment (NateCastleOIT) [inline TKGame/Content/Weapons/Sword.cs:28]: Are we only going to have a single weapon instance ever?

File: TKGame/Debug.cs [hunk 2]
```diff
@@ -63,19 +68,17 @@ public static void Initialize()
         {
             DebugMode = false;
             KeyboardGrid = new Grid();
+            spacer1 = new Label();
+            spacer2 = new Label();
```
Comment (NateCastleOIT) [inline TKGame/Debug.cs:72]: These could be a bit more descriptive

File: TKGame/Debug.cs [hunk 3]
```diff
@@ -88,7 +91,7 @@ public static void Initialize()
         /// <summary>
         /// Load/Stylize all debug UI elements
         /// </summary>
-        public static void LoadContent()
+        public static void LoadContent(VerticalStackPanel VSP)
```
Comment (NateCastleOIT) [inline TKGame/Debug.cs:94]: Why was this call changed? I don't think we want to be passing every stack panel in like this

File: TKGame/WeaponSystem.cs [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+        /// <summary>
+        /// Update debug UI elements
+        /// </summary>
+        public static void Update()
+        {
+            // TODO: Setup color change with active weapon
+            swordLabel.Text = swordText;
+            spearLabel.Text = spearText;
```
Comment (NateCastleOIT) [inline TKGame/WeaponSystem.cs:109]: I'm not sure this needs to be done every Update as it seems more like something that will only happen once and then should always appear

**Review by NateCastleOIT — APPROVED** [review]
A few questions I commented but otherwise looks fine

File: TKGame/Content/Weapons/Sword.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TKGame.Content.Weapons
```
Comment (aaronw84) [inline TKGame/Content/Weapons/Sword.cs:9]: This is a good framework for the weapons. For now having the sword be implemented by the weapons is fine, like what I did for Item. We'll need to make this abstract in the future to have concrete weapons inherit from it.

File: TKGame/Content/Weapons/Weapon.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TKGame.Content.Weapons
```
Comment (aaronw84) [inline TKGame/Content/Weapons/Weapon.cs:9]: This is exactly what I plan to do with the Item class. Make it an abstract class to use the strategy pattern.

**Review by aaronw84 — APPROVED** [review]

**Pull request merged**